lib/src/test/java/io/ably/lib/test/rest/RestAuthAttributeTest.java

```diff
@@ -0,0 +1,95 @@
+import io.ably.lib.types.ClientOptions;
+
+import static io.ably.lib.rest.Auth.AuthOptions;
+import static io.ably.lib.rest.Auth.TokenCallback;
+import static io.ably.lib.rest.Auth.TokenDetails;
+import static io.ably.lib.rest.Auth.TokenParams;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+/**
+ * Created by VOstopolets on 9/3/16.
+ */
+public class RestAuthAttributeTest {
+
+    private static AblyRest ably;
+
+    @BeforeClass
+    public static void setUpBeforeClass() throws Exception {
+        Setup.TestVars testVars = Setup.getTestVars();
+        ClientOptions opts = testVars.createOptions(testVars.keys[0].keyStr);
+        opts.clientId = "defaultClientId";
+        ably = new AblyRest(opts);
+    }
+
+    /**
+     * Stores the AuthOptions and TokenParams arguments as defaults
+     * for subsequent authorisations with the exception of the attributes:
+     * {@link TokenParams#timestamp}, {@link AuthOptions#queryTime}, {@link AuthOptions#force}
+     * <p>
+     * Spec: RSA10g
+     * </p>
+     */
+    @Test
+    public void auth_stores_options_params() {
+        try {
+            /* init custom TokenParams */
+            final Capability capability1 = new Capability();
+            capability1.addResource("testchannel", "subscribe");
+            TokenParams tokenParams = new TokenParams() {{
+                ttl = 4000L;
+                clientId = "firstClientId";
+                capability = capability1.toString();
+            }};
+
+            /* init custom AuthOptions */
+            final Setup.TestVars testVars = Setup.getTestVars();
+            AuthOptions authOptions = new AuthOptions() {{
+                authCallback = new TokenCallback() {
+                    private AblyRest ably = new AblyRest(testVars.createOptions(testVars.keys[0].keyStr));
+
+                    @Override
+                    public Object getTokenRequest(TokenParams params) throws AblyException {
+                        return ably.auth.requestToken(params, null);
+                    }
+                };
+                key = testVars.keys[1].keyStr;
+            }};
+
+            /* authorise with custom options */
+			TokenDetails tokenDetails1 = ably.auth.authorise(authOptions, tokenParams);
+
+			/* wait until token expires */
+			try {
+				Thread.sleep(5000L);
+			} catch(InterruptedException ie) {}
+
+			/* authorise with default options */
+			TokenDetails tokenDetails2 = ably.auth.authorise(null, null);
+
+			/* Verify that,
+			 * tokenDetails1 and tokenDetails2 aren't null,
+			 * new token has to be issued,
+			 * capability and clientId for different TokenDetails are the same */
+			assertNotNull(tokenDetails1);
+			assertNotNull(tokenDetails2);
+			assertNotEquals(tokenDetails1.token, tokenDetails2.token);
+			assertEquals(tokenDetails1.capability, tokenDetails2.capability);
```

Comment:
I appreciate this probably works, but if tokenDetails1.capability was invalid then the second test here would be pointless as at no point are you checking that the capability matches `capability`.  Same applies to clientID, you never check that it is ever `firstClientId` so it could feasibly be `defaultClientId` and this test would pass.  Please update your tests to assert the expectations after the first token is issued.